src/Nethermind/Nethermind.State.Test/StorageProviderTests.cs

```diff
@@ -185,20 +185,17 @@ public void Commit_trees_clear_caches_get_previous_root()
         WorldState storageProvider = BuildStorageProvider(ctx);
         storageProvider.Set(new StorageCell(ctx.Address1, 1), _values[1]);
         storageProvider.Commit(Frontier.Instance);
-
-        ctx.StateProvider.Commit(Frontier.Instance);
-        ctx.StateProvider.CommitTree(0);
+        storageProvider.Commit(Frontier.Instance);
+        storageProvider.CommitTree(0);
 
         // block 2
         Hash256 stateRoot = ctx.StateProvider.StateRoot;
         storageProvider.Set(new StorageCell(ctx.Address1, 1), _values[2]);
         storageProvider.Commit(Frontier.Instance);
-        ctx.StateProvider.Commit(Frontier.Instance);
```

Comment:
`storageProvider` and `ctx.StateProvider` are the same, so using both is confusing